### PR TITLE
Add rank threshold

### LINKED ISSUE #544: [v1.9] Support `rankingScoreThreshold` search parameter
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/301)*

This is related to a newly introduced feature in [Meilisearch v1.9.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.9.0): the `rankingScoreThreshold` search parameter.

Refer to [docs for more information](https://www.meilisearch.com/docs/reference/api/search#ranking-score-threshold).

- [ ] Ensure the library can accept the new `rankingScoreThreshold` search parameter.
- [ ] ⚠️ Add integration tests ⚠️
- [ ] Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `search_parameter_reference_ranking_score_threshold_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/04f26a35771305378ed0a506f4d60ef06d07bd7e/.code-samples.meilisearch.yaml#L1344

### PR DESCRIPTION
**Pull request by danFbach**
# Pull Request

## Related issue
Fixes #544 

## What does this PR do?
- adds support for utilizing `RankingScoreThreshold` `SearchQuery` parameter.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

**Comment by danFbach** [comment]
Is it necessary to enforce the threshold range of `0.0 - 1.0` locally, or can we just allow meilisearch to ignore/reject when the value falls outside that range?

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -782,3 +782,10 @@ distinct_attribute_guide_distinct_parameter_1: |-
     Distinct = "sku"
   };
   await client.Index("products").SearchAsync<Product>("white shirt", params);
+search_parameter_reference_ranking_score_threshold_1: |-
+  var params = new SearchQuery()
+  {
+    RankingScoreThreshold = 0.2M
+  };
+  await client.Index("INDEX_NAME").SearchAsync<T>("badman", params);
+
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml]:
```suggestion
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -782,3 +782,9 @@ distinct_attribute_guide_distinct_parameter_1: |-
     Distinct = "sku"
   };
   await client.Index("products").SearchAsync<Product>("white shirt", params);
+search_parameter_reference_ranking_score_threshold_1: |-
+  var params = new SearchQuery()
+  {
+    RankingScoreThreshold = 0.2M
+  };
+  await client.Index("INDEX_NAME").SearchAsync<T>("badman", params);
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:790]:
```suggestion
  await client.Index("INDEX_NAME").SearchAsync<T>("badman", params);
```

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -782,3 +782,9 @@ distinct_attribute_guide_distinct_parameter_1: |-
     Distinct = "sku"
   };
   await client.Index("products").SearchAsync<Product>("white shirt", params);
+search_parameter_reference_ranking_score_threshold_1: |-
+  var params = new SearchQuery()
+  {
+    RankingScoreThreshold = 0.2M
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:788]: why 0.2M? and not `0.2`, cf https://github.com/meilisearch/documentation/blob/04f26a35771305378ed0a506f4d60ef06d07bd7e/.code-samples.meilisearch.yaml#L1350

File: .code-samples.meilisearch.yaml (hunk 2 above)
Comment (curquiza) [inline .code-samples.meilisearch.yaml]: Can you remove the "no new line" at the end of the line?

Comment (curquiza) [reply .code-samples.meilisearch.yaml]: we follow the editorconfig rules

**Review by curquiza — CHANGES_REQUESTED** [review]

**Comment by curquiza** [comment]
@danFbach 

Looks like there is an issue with the lining (cf CI)

Can you fix this test?

File: .code-samples.meilisearch.yaml (hunk 3 above)
Comment (danFbach) [reply .code-samples.meilisearch.yaml:788]: c# requires `M` when setting a decimal value. `0.2` is interpreted as a `double` and `0.2M` is a `decimal`.
ex: 
image

**Review by curquiza — APPROVED** [review]
Awesome!

thank you

bors merge

**Pull request merged**